eodag/rest/utils.py

```diff
@@ -699,6 +739,31 @@ def get_stac_item_by_id(
         return None
 
 
+def _retrieve_from_cache(
+    provider: Optional[str], product_type: str, item_id: str
+) -> Optional[EOProduct]:
+    if use_redis:
+        if provider:
+            cache_key = f"{product_type}_{provider}_{item_id}"
+        else:
+            cache_key = f"{product_type}_{item_id}"
```

Comment:
cf my previous comment

Reply:
cache keys now always contain provider; if keys for several providers available and no provider is given, the highest priority provider is used